pandas/core/reshape/concat.py

```diff
@@ -362,15 +362,30 @@ def get_result(self):
 
             # stack blocks
             if self.axis == 0:
+                name = com._consensus_name_attr(self.objs)
+
                 # concat Series with length to keep dtype as much
                 non_empties = [x for x in self.objs if len(x) > 0]
+
+                # check if all series are of the same block type:
```

Comment:
this is the purpose of concat_compat, why are you re-writing?

Reply:
`concat_compat` works on the level of *values*, here I want to work on the level of *blocks*. I could also try to do that in `concat_compat`, but I am not sure I can fully switch that to using blocks (eg for concatting index it still needs to work on values)

Reply:
yeah we just pushed the sparse handling for this down into the blocks to make ``.unstack`` cleaner. Its not that hard actually. you would just have a BlockManager.concat which does the work for you (e.g. give it a list of other block managers).

Reply:
right seem my comment. This would be ideal to actually do *inside* the BlockManger (not objecting to the actual code, more of its location).

Reply:
For series this would be SingleBlockManager then? 

For me that would be OK to move this there, but I in the SingleBlockManager.concat, I would still need to dispatch to the actual Blocks (as I now have the methods on the individual blocks), as my original goal is enabling external blocks to override how their values are concatenated.

Reply:
sure that's fine, your code would be pretty similar. the idea is you just have a Block.concat that works for mostl cases and case override as needed (e.g. in your custom block).

Reply:
yeah, the sparse unstack PR is indeed a good example for this

Reply:
So I did an attempt to move the logic inside SingleBlockManager, see last commit. Not fully sure I find it cleaner (as eg the new_axis is passed into the function, as that logic is currently quite ingrained into the `_Concatenator` class to calculate the new axis values)